src/Features/CSharp/Portable/CodeFixes/AddParameterXmlDocumentation/AddParameterXmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,316 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.AddParameterXmlDocumentation
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddParameterXmlDocumentaion), Shared]
+    internal partial class AddParameterXmlDocumentationCodeFixProvider : CodeFixProvider
+    {
+        // Parameter 'parameter' has no matching param tag in the XML comment for 'parameter' (but other parameters do)
+        private const string CS1573 = nameof(CS1573);
+        // Type parameter 'type parameter' has no matching typeparam tag in the XML comment on 'type' (but other type parameters do)
+        private const string CS1712 = nameof(CS1712);
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CS1573, CS1712);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        private BaseParameterListSyntax FindFirstAncestorParameterItem(SyntaxNode node)
+        {
+            for (; node != null; node = node.Parent)
+            {
+                if (node is BaseMethodDeclarationSyntax)
+                {
+                    return ((BaseMethodDeclarationSyntax)node).ParameterList;
+                }
+                if (node is IndexerDeclarationSyntax)
+                {
+                    return ((IndexerDeclarationSyntax)node).ParameterList;
+                }
+                if (node is DelegateDeclarationSyntax)
+                {
+                    return ((DelegateDeclarationSyntax)node).ParameterList;
+                }
+            }
+            return null;
+        }
+
+        // Can't use the generic GetAncestor method because structs and classes can nest each other.
+        private TypeParameterListSyntax FindFirstAncestorTypeParameterItem(SyntaxNode node)
+        {
+            for (; node != null; node = node.Parent)
+            {
+                if (node is MethodDeclarationSyntax)
+                {
+                    return ((MethodDeclarationSyntax)node).TypeParameterList;
+                }
+                if (node is ClassDeclarationSyntax)
+                {
+                    return ((ClassDeclarationSyntax)node).TypeParameterList;
+                }
+                if (node is StructDeclarationSyntax)
+                {
+                    return ((StructDeclarationSyntax)node).TypeParameterList;
+                }
+                if (node is InterfaceDeclarationSyntax)
+                {
+                    return ((InterfaceDeclarationSyntax)node).TypeParameterList;
+                }
+                if (node is DelegateDeclarationSyntax)
+                {
+                    return ((DelegateDeclarationSyntax)node).TypeParameterList;
+                }
```

Comment:
There is an existing extension method you can use for this: http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Extensions/MemberDeclarationSyntaxExtensions.cs,8c01aa4c9978cdbd